sql/core/src/test/scala/org/apache/spark/sql/execution/command/v1/ShowTablesSuite.scala

```diff
@@ -165,4 +154,63 @@ class ShowTablesSuite extends ShowTablesSuiteBase with CommandSuiteBase {
       }
     }
   }
+
+  override protected def extendedPartInNonPartedTableError(
+      catalog: String,
+      namespace: String,
+      table: String): (String, Map[String, String]) = {
+    ("_LEGACY_ERROR_TEMP_1251",
```

Comment:
how hard it is to unify this error between v1 and v2 tables?

Reply:
- _LEGACY_ERROR_TEMP_1231(QueryCompilationErrors#invalidPartitionColumnKeyInTableError), there are approximately `13 `code point to call it:
   <img width="465" alt="image" src="https://github.com/apache/spark/assets/15246973/7d61f88a-c567-43a5-b7a9-5751ac61613d">
https://github.com/apache/spark/blob/1359c1327345efdf9a35c46a355b5f928ac33e6d/common/utils/src/main/resources/error/error-classes.json#L4815-L4819

- _LEGACY_ERROR_TEMP_1251(QueryCompilationErrors#actionNotAllowedOnTableSincePartitionMetadataNotStoredError), there are approximately `8 `code point to call it:
   <img width="711" alt="image" src="https://github.com/apache/spark/assets/15246973/0a18def1-b604-44ee-990f-ed1fdaa2d7a0">
https://github.com/apache/spark/blob/1359c1327345efdf9a35c46a355b5f928ac33e6d/common/utils/src/main/resources/error/error-classes.json#L4860-L4864

Because there are many scopes involved, in order to reduce the interference of logic on this PR, I suggest doing `the merging or unification` in a new separate PR. 
Do you think this is appropriate?

Reply:
make senses, let's do it in followup